src/freecoin/core.clj

```diff
@@ -174,18 +174,18 @@
 
 ;; For running from the repl
 (defn start []
-  (alter-var-root #'app-state (comp launch connect-db)))
+  (swap! app-state (comp launch connect-db)))
 
 (defn stop []
-  (alter-var-root #'app-state (comp disconnect-db halt)))
+  (swap! app-state (comp disconnect-db halt)))
 
 ;; For running using lein-ring server
-(defonce lein-ring-handler nil)
+(defonce lein-ring-handler (atom nil))
 
 (defn lein-ring-init []
   (prn "lein-ring-init")
-  (alter-var-root #'app-state connect-db)
```

Comment:
many thanks! now finally I understand what was `alter-var-root`. this code was Duncan's and I couldn't understand it well...